[Bug] Crash when opening statistics

I have ankidroid 2.15.5beta1
My phone is a google pixel 4a 5g on android 11


I tried restarting my phone and it did nothing. Everything else seems to work fine.



edit : i don't know what happened but now it works? i just got to a new day and it's not crashing but it might happen again once i review cards?

Likely: https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/cc588ffa-0511-4eba-b1f1-cab6f363fc52

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$3.done(AsyncTask.java:354)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.IndexOutOfBoundsException: Index: 25, Size: 24
at java.util.ArrayList.set(ArrayList.java:453)
at com.ichi2.libanki.stats.Stats.calculateBreakdown(Stats.java:22)
at com.ichi2.anki.stats.ChartBuilder.calcStats(ChartBuilder.java:6)
at com.ichi2.anki.stats.ChartBuilder.renderChart(ChartBuilder.java:1)
at com.ichi2.anki.stats.AnkiStatsTaskHandler$CreateChartTask.doInBackgroundSafe(AnkiStatsTaskHandler.java:10)
at com.ichi2.anki.stats.AnkiStatsTaskHandler$CreateChartTask.doInBackgroundSafe(AnkiStatsTaskHandler.java:1)
at com.ichi2.anki.stats.StatsAsyncTask.doInBackground(StatsAsyncTask.java:2)
at com.ichi2.anki.stats.StatsAsyncTask.doInBackground(StatsAsyncTask.java:1)
at android.os.AsyncTask$2.call(AsyncTask.java:333)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
... 4 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/stats/Stats.java
@@ -892,7 +892,7 @@ public class Stats {
                     .query(query)) {
             while (cur.moveToNext()) {
                 double[] hourData = new double[] { cur.getDouble(0), cur.getDouble(1), cur.getDouble(2) };
-                list.set((int)hourData[0], hourData);
+                list.set(((((int)hourData[0] % 24) + 24) % 24), hourData); // Force the data to be positive int in 0-23 range
             }
         }
 

